core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -0,0 +1,529 @@
+package io.aiven.klaw.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.aiven.klaw.config.ManageDatabase;
+import io.aiven.klaw.dao.Acl;
+import io.aiven.klaw.dao.Env;
+import io.aiven.klaw.dao.KwClusters;
+import io.aiven.klaw.dao.Topic;
+import io.aiven.klaw.helpers.HandleDbRequests;
+import io.aiven.klaw.model.AclInfo;
+import io.aiven.klaw.model.ApiResultStatus;
+import io.aiven.klaw.model.KafkaClustersType;
+import io.aiven.klaw.model.TopicHistory;
+import io.aiven.klaw.model.TopicInfo;
+import io.aiven.klaw.model.TopicOverview;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class TopicOverviewService {
+
+  public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  public static final ObjectWriter WRITER_WITH_DEFAULT_PRETTY_PRINTER =
+      OBJECT_MAPPER.writerWithDefaultPrettyPrinter();
+  @Autowired ManageDatabase manageDatabase;
+  @Autowired ClusterApiService clusterApiService;
+  @Autowired private CommonUtilsService commonUtilsService;
+  private final MailUtils mailService;
+
+  public TopicOverviewService(MailUtils mailService) {
+    this.mailService = mailService;
+  }
+
+  public TopicOverview getTopicOverview(String topicNameSearch) {
+    log.debug("getAcls {}", topicNameSearch);
+    String userDetails = getUserName();
```

Comment:
this line should be moved **after** this `if`
```java
if (topicNameSearch != null) {
      topicNameSearch = topicNameSearch.trim();
    } else {
      return null;
    }
```
to avoid useless retrieval of data from db in case returning of null

Reply:
agree, updated.